python/TestHarness/RunParallel.py

```diff
@@ -143,6 +154,13 @@ def run(self, tester, command, recurse=True, slot_check=True):
         self.jobs[job_index] = (p, command, tester, clock(), f, slots)
         self.slots_in_use = self.slots_in_use + slots
 
+    def addJobID(self, file, job_id):
+        with open(file, 'r') as f:
+            tmp_contents = f.read()
+        with open(file, 'w') as f:
+            tmp_contents = tmp_contents.replace( '<JOB_ID>', str(job_id) )
+            f.write(tmp_contents)
+
```

Comment:
Open the existing file a second time... and replace <JOB_ID> with a list of dependency job id/s. There should be a way to only deal with this file once.